extras/check_code_style.sh

```diff
@@ -0,0 +1,215 @@
+    --clang-format=exe   Enter the executable that is used for clang-format.
+                         (default=clang-format)
+    --vera++=exe         Enter the executable that is used for vera++.
+                         (default=vera++)
+EOF
+
+    exit $1
+}
+
+#
+# bail_out message
+#
+bail_out ()
+{
+    echo "$1"
+    exit 1
+}
+
+while test $# -gt 0 ; do
+    case "$1" in
+        --help)
+            usage 0
+            ;;
+        --incremental)
+            INCREMENTAL=true
+            ;;
+        --file=*)
+            ONLY_FILE="$( echo "$1" | sed 's/^--file=//' )"
+            if [ ! -f $ONLY_FILE ]; then
+              bail_out "This is not a valid file: $ONLY_FILE"
+            fi
+            ;;
+        --git-start=*)
+            START_SHA="$( echo "$1" | sed 's/^--git-start=//' )"
+            ;;
+        --git-end=*)
+            END_SHA="$( echo "$1" | sed 's/^--git-end=//' )"
+            ;;
+        --nest-src=*)
+            NEST_SRC="$( echo "$1" | sed 's/^--nest-src=//' )"
+            ;;
+        --cppcheck=*)
+            CPPCHECK="$( echo "$1" | sed 's/^--cppcheck=//' )"
+            ;;
+        --vera++=*)
+            VERA="$( echo "$1" | sed 's/^--vera++=//' )"
+            ;;
+        --clang-format=*)
+            CLANG_FORMAT="$( echo "$1" | sed 's/^--clang-format=//' )"
+            ;;
+        *)
+            usage 1 "$1"
+            ;;
+    esac
+    shift
+done
+
+#
+# sed has different syntax for extended regular expressions
+# on different operating systems:
+# BSD: -E
+# other: -r
+#
+EXTENDED_REGEX_PARAM=r
+/bin/sh -c "echo 'hello' | sed -${EXTENDED_REGEX_PARAM} 's/[aeou]/_/g' "  >/dev/null 2>&1 || EXTENDED_REGEX_PARAM=E
+
+echo "Executing check_code_style.sh"
+echo "Tooling:"
+
+# check vera++ is set up appropriately
+$VERA $NEST_SRC/nest/main.cpp >/dev/null 2>&1 || usage 1 "Executable $VERA for vera++ is not working!"
+$VERA --profile nest $NEST_SRC/nest/main.cpp >/dev/null 2>&1 ||  bail_out "No profile called nest for vera++ installed. See https://nest.github.io/nest-simulator/coding_guidelines_c++#vera-profile-nest"
+echo "vera++ version: `vera++ --version`"
+
+# check cppcheck 1.69 is installed correctly
+$CPPCHECK --enable=all --inconclusive --std=c++03 $NEST_SRC/nest/main.cpp >/dev/null 2>&1 || usage 1 "Executable $CPPCHECK for cppcheck is not working!"
+cppcheck_version=`$CPPCHECK --version | sed 's/^Cppcheck //'`
+echo "cppcheck version: $cppcheck_version"
+if [[ "x$cppcheck_version" != "x1.69" ]]; then
+  bail_out "Require cppcheck version 1.69. Not $cppcheck_version ."
+fi
+
+# clang-format is installed correctly
+$CLANG_FORMAT -style=$NEST_SRC/.clang-format $NEST_SRC/nest/main.cpp >/dev/null 2>&1 || usage 1 "Executable $CLANG_FORMAT for clang-format is not working!"
+clang_format_version=`$CLANG_FORMAT --version | sed -${EXTENDED_REGEX_PARAM} 's/^.*([0-9]\.[0-9])\..*/\1/'`
+echo "clang-format version: $clang_format_version"
+if [[ "x$clang_format_version" != "x3.6" ]]; then
```

Comment:
According to developer space, we require 3.6 *or later*. Why do you require precisely 3.6 here?